lib/moon/design/form/switch.ex

```diff
@@ -0,0 +1,66 @@
+defmodule Moon.Design.Form.Switch do
+  @moduledoc false
+
+  use Moon.StatefulComponent
+
+  alias Moon.Design.Switch
+
+  prop(field, :atom, from_context: {Surface.Components.Form.Field, :field})
+  prop(form, :form, from_context: {Surface.Components.Form, :form})
+
+  prop(is_switched, :boolean, default: false)
+  prop(options, :list, default: [])
+  prop(value, :any)
+  prop(name, :string)
+
+  prop(disabled, :boolean, default: false)
+  prop(size, :string, values: ["2xs", "xs", "sm"], default: "sm")
+  prop(on_bg_color, :css_class, default: "bg-piccolo")
+  prop(off_bg_color, :css_class, default: "bg-beerus")
+  prop(class, :css_class)
+  prop(test_id, :string)
+
+  slot(on_icon)
+  slot(off_icon)
+
+  def render(assigns) do
+    ~F"""
+    <div>
+      <input
+        type="checkbox"
+        name={@name}
+        value={@value}
+        checked={@is_switched}
+        class="hidden"
+        :on-click="toggle_switch"
```

Comment:
why do we need this event here?

Reply:
This event toggles the checkbox. Should it be done in some other way?

Reply:
but this checkbox is hidden, so the event will never occur as i understand